p751toolbox/field_decl.go

```diff
@@ -34,9 +38,30 @@ func fp751Mul(z *fp751X2, x, y *Fp751Element)
 
 // Perform Montgomery reduction: set z = x R^{-1} (mod 2*p).
 // Destroys the input value.
+var fp751MontgomeryReduce func(z *Fp751Element, x *fp751X2)
+
+//go:noescape
+func fp751MontgomeryReduceBMI2ADX(z *Fp751Element, x *fp751X2)
+
 //go:noescape
-func fp751MontgomeryReduce(z *Fp751Element, x *fp751X2)
+func fp751MontgomeryReduceBMI2(z *Fp751Element, x *fp751X2)
+
+//go:noescape
+func fp751MontgomeryReduceFallback(z *Fp751Element, x *fp751X2)
 
 // Reduce a field element in [0, 2*p) to one in [0,p).
 //go:noescape
 func fp751StrongReduce(x *Fp751Element)
+
+// Choose the fastest variant depending on CPU capabilities.
+func init() {
```

Comment:
That's better, but now code coverage has dropped. Obviously, untested code isn't acceptable. Can you make sure that we run tests against all flavors on a CPU with MULX and ADOX?